src/Components/Blueprints/BuildImagesButton.tsx

```diff
@@ -1,20 +1,160 @@
+      await buildBlueprint({
+        id: selectedBlueprintId,
+        body: {
+          image_types: blueprintImageType?.filter(
+            (target) => !selectedTargets?.includes(target)
+          ),
+        },
+      });
+    }
+  };
+  const [isOpen, setIsOpen] = useState(false);
+  const onToggleClick = () => {
+    setIsOpen(!isOpen);
+  };
+  const { data: blueprintDetails } = useGetBlueprintQuery(
+    selectedBlueprintId ? { id: selectedBlueprintId } : skipToken
+  );
+  const blueprintImageType = blueprintDetails?.image_requests.map(
+    (image) => image.image_type
+  );
+
+  const onSelect = (
+    _event: React.MouseEvent<Element, MouseEvent>,
+    itemId: number
+  ) => {
+    const imageType = blueprintImageType?.[itemId];
+
+    if (imageType && selectedTargets?.includes(imageType)) {
+      setSelectedTargets(
+        selectedTargets.filter((target) => target !== imageType)
+      );
+    } else if (imageType) {
+      setSelectedTargets([...selectedTargets, imageType]);
+    }
+  };
+
+  return (
+    <Dropdown
+      isOpen={isOpen}
+      onOpenChange={(isOpen: boolean) => setIsOpen(isOpen)}
+      toggle={(toggleRef: React.Ref<MenuToggleElement>) => (
+        <MenuToggle
+          variant="primary"
+          ref={toggleRef}
+          isFullWidth
+          onClick={onToggleClick}
+          isExpanded={isOpen}
+          splitButtonOptions={{
+            variant: 'action',
+            items: [
+              <MenuToggleAction
+                key="split-action"
+                onClick={onBuildHandler}
+                id="wizard-build-image-btn"
+                isDisabled={!selectedBlueprintId}
+              >
+                <Flex display={{ default: 'inlineFlex' }}>
+                  {imageBuildLoading && (
+                    <FlexItem>
+                      <Spinner
+                        style={
+                          {
+                            '--pf-v5-c-spinner--Color': '#fff',
+                          } as React.CSSProperties
+                        }
+                        isInline
+                        size="md"
+                      />
+                    </FlexItem>
+                  )}
+                  <FlexItem>
+                    {selectedTargets.length !== 0
+                      ? 'Build selected'
+                      : 'Build Image'}
+                  </FlexItem>
+                </Flex>
+              </MenuToggleAction>,
+            ],
+          }}
+        ></MenuToggle>
+      )}
+    >
+      <Menu onSelect={onSelect} selected={blueprintImageType}>
+        <MenuContent>
+          <MenuList>
+            {blueprintImageType?.map((imageType, index) => (
+              <MenuItem
+                key={imageType}
+                hasCheckbox
+                itemId={index}
+                isSelected={
+                  !selectedTargets || !selectedTargets.includes(imageType)
+                }
+              >
+                {imageType
```

Comment:
This might be better extracted to a separate function.
Might be ok to just have it in this file for now, but ideal would be some utilities/const file.
```js
export const humanReadableImageTarget = (target) =>
  switch(target) {
    case 'aws':
     return 'Amazon Web Services (AWS)';
  .....
```

I also wonder if we don't already have it, we need to have targets human translated.
EDIT: we do not have anything like it yet as far as I could find, so just having such function here localy makes sense :)

Reply:
Extracting the replace logic into separate function sounds good. I have an idea on how I'd like to deal with the mapping throughout the code base so I'd open a follow up PR once this is merged.